### DISCUSSION TITLE
Rename / Edit feature for Console Resources

### DISCUSSION THREAD

**Discussion by flower-of-the-bridges** [mia-platform/community#252] [Ideas]
### Which area of this project could be most improved?

Console

### Description

A nice feature to add to each section (endpoint/microservice/config map) is to add the following operations:

- duplicate 
- rename

Here are some use cases that could be helpful to understand the possible scenarios where such features could be useful.

## Endpoint duplication

I have an endpoint `/example`: If I want to make an endpoint with similar methods and user group permission, I need to create a new endpoint and copy/paste the resources manually. In this case, a `duplicate` button could simplify the process

## Endpoint Rename  

I have an endpoint called `/xample`: If I want to rename the endpoint into `/example`, at the moment I need to delete it and re-create it from scratch. In this case, a `edit` feature that can enable to rename the endpoint could simplify the process

## Microservice duplication

I have a microservice called `example-microservice`: If I want to make a microservice with similar envs and config maps, I need to create a new endpoint and copy/paste manually both envs and config maps. In this case, a `duplicate` button could simplify the process

## Microservice Rename  

I have a microservice called `xample-microservice`: If I want to rename the endpoint into `example-microservice`, I need to delete it and re-create it from scratch. In this case, an `edit` feature that can enable to rename the endpoint could simplify the process

## Config Map File duplication

I have a config map named `example-config` with a file `example.js`: If I want to create a new file having the same content, I need to create a new file and copy/paste manually its button. In this case, a `duplicate` button could simplify the process

## Config Map File Rename  

I have a config map named `example-config` with a file `xample.js`: If I want to rename the file into `example.js`, I need to delete it and re-create it from scratch. In this case, an `edit` feature that can enable to rename the endpoint could simplify the process

**Comment by fredmaggiowski** [comment]
Hey, thank you for the feedback, i agree that endpoint renaming and duplication could be a nice feature to have.

I don't think workload duplication is, however we could improve the environment variables imoort/export feature to make it easier to copy them from one service to another, same may apply to configmap file, even though they are already easy to copied over without the need for specific implementations

@albertotessarotto let's track these feedback down, they are easy to implement and provide lots of value

**Comment by akelity** [comment]
This feature has been positively evaluated and may be considered for inclusion in the roadmap in the future, depending on priorities and overall specific needs.
